source/Ship.cpp

```diff
@@ -601,6 +625,20 @@ void Ship::Load(const DataNode &node)
 // loaded yet. So, wait until everything has been loaded, then call this.
 void Ship::FinishLoading(bool isNewInstance)
 {
+	// Automatically create an explosion for this ship if requested but
+	// only if the ship does not already have a defined explosion.
+	if(autoExplosion && !baseAttributes.IsWeapon())
+	{
+		double baseShields = baseAttributes.Get("shields");
+		double baseHull = baseAttributes.Get("hull");
+		const double scale = (baseShields + baseHull) * autoExplosion->baseMult;
+		const double radius = scale * autoExplosion->radiusMult;
+		const double shieldDamage = scale * autoExplosion->shieldMult;
+		const double hullDamage = scale * autoExplosion->hullMult;
+		const double hitForce = scale * autoExplosion->forceMult;
```

Comment:
Marking these variables as const doesn't make much sense.
```suggestion
		double scale = (baseShields + baseHull) * autoExplosion->baseMult;

		double radius = scale * autoExplosion->radiusMult;
		double shieldDamage = scale * autoExplosion->shieldMult;
		double hullDamage = scale * autoExplosion->hullMult;
		double hitForce = scale * autoExplosion->forceMult;
```

Reply:
Why?

Reply:
> Why?

The `const`s serve no purpose and it's (subjectively) nicer to read by leaving them out.